scripts/profiler/events.py

```diff
@@ -47,6 +46,37 @@ def __init__(self, submit, start_time, end_time):
         self.proc_start_time = start_time
         self.proc_end_time = end_time
 
+    def serialize(self):
+        si = StringIO()
+        sw = csv.writer(si, delimiter=',')
+        sw.writerow(list((self.submit.type_id,
+                    self.submit.timestamp,
+                    self.proc_start_time,
+                    self.proc_end_time,
+                    *self.submit.data)))
+        return si.getvalue().strip('\r\n')
+
+    @staticmethod
+    def deserialize(scsv):
+        f = StringIO(scsv)
+        reader = csv.reader(f, delimiter=',')
+        #There should be only one row
+        for row in reader:
+            return TrackedEvent(Event(int(row[0]), float(row[1]), list(row[4:])),
+                                None if row[2]=='' else float(row[2]),
+                                None if row[3]=='' else float(row[3]))
+
+    def write_event_to_file(self, csvfile):
+        fieldnames = ['type_id', 'timestamp', 'data', 'proc_start_time', 'proc_end_time']
+        wr = csv.DictWriter(csvfile, delimiter=',',
```

Comment:
Should we create csv.DictWriter everytime we write a single event?
Maybe we could only provide API for serialization and fieldnames in this class ?